### PR TITLE
Stats Views widget setup

### PR DESCRIPTION
**Pull request by planarvoid**
This PR introduces classes necessary for the Views widget. I've also added the configure activity (it's not fully functional yet). Most of the stuff here doesn't need to be reviewed because it will be completely rewritten anyway but I didn't want to remove the classes since it took a lot of effort to make everything work. The Widgets API is dreadful. 

To test:
* Go to your Home and try to add a Widget
* Scroll to the Wordpress and add the new widget
* You see a new configure activity
* The Site and Color button don't work yet (TODO crashes the app if you click on them)
* The "Add Widget" button is disabled

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsViewsWidget.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import android.os.Bundle
+import android.widget.RemoteViews
+import org.wordpress.android.R
+import org.wordpress.android.R.id
+
+/**
+ * Implementation of App Widget functionality.
+ * App Widget Configuration implemented in [StatsViewsWidgetConfigureActivity]
+ */
+class StatsViewsWidget : AppWidgetProvider() {
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsViewsWidget.kt]: This class will completely change so it doesn't need to be reviewed now.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsListProvider.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import android.appwidget.AppWidgetManager
+import android.content.Context
+import android.content.Intent
+import android.widget.RemoteViews
+import android.widget.RemoteViewsService.RemoteViewsFactory
+import org.wordpress.android.R.layout
+import org.wordpress.android.fluxc.model.stats.time.VisitsAndViewsModel
+
+class ViewsListProvider(val context: Context, intent: Intent) : RemoteViewsFactory {
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsListProvider.kt]: This class will completely change so it doesn't need to be reviewed now.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    private val mutableWidgetAdded = MutableLiveData<Event<WidgetAdded>>()
+    val widgetAdded: LiveData<Event<WidgetAdded>> = mutableWidgetAdded
+    private var appWidgetId: Int = -1
+
+    fun start(appWidgetId: Int) {
+        this.appWidgetId = appWidgetId
+        val colorModeId = appPrefsWrapper.getAppWidgetColorModeId(appWidgetId)
+        if (colorModeId >= 0) {
+            mutableViewMode.postValue(ViewMode.values()[colorModeId])
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetViewModel.kt]: What do you think about using some sort of ID for `ViewMode`? This way we can protect ourself from a change in their order and content.

**Review by khaykov — APPROVED** [review]
I assume the majority of code in fragment and ViewModel is also WIP, but as a "skeleton" this works! I'll make a mental note to comb through everything when are about to merge feature into develop.

**Pull request merged**